sqlitecluster/SQLiteNode.h

```diff
@@ -205,10 +213,32 @@ class SQLiteNode : public STCPNode {
     chrono::steady_clock::time_point _lastNetStatTime;
 
     // Handler for transaction messages.
-    void handleBeginTransaction(Peer* peer, const SData& message);
-    void handleCommitTransaction(Peer* peer, const SData& message);
-    void handleRollbackTransaction(Peer* peer, const SData& message);
+    void handleBeginTransaction(SQLite& db, Peer* peer, const SData& message);
+    void handleCommitTransaction(SQLite& db, Peer* peer, const uint64_t commandCommitCount, const string& commandCommitHash);
+    void handleRollbackTransaction(SQLite& db, Peer* peer, const SData& message);
 
     WallClockTimer _syncTimer;
-    uint64_t _handledCommitCount;
+    atomic<uint64_t> _handledCommitCount;
+
+    // List of threads that are handling replication. These are transient and only exist while FOLLOWING.
+    list<thread> _replicationThreads;
+
+    // State variable that indicates when the above threads should quit.
+    atomic<bool> _replicationThreadsShouldExit;
+
+    // Mutex that guards the map of hashes for transactions currently being replicated.
+    mutex _replicationHashMutex;
+    map<uint64_t, string> _replicationHashes;
```

Comment:
I suggest renaming this to `_replicationHashesToCommit`, because when a hash is put here, it authorizes it to be committed.

Reply:
Why is this a `map`, when `_rollbackHashes` is a `set`?  It seems a set is better.  Indeed, I like how commits are being identified by hash versus commit count, because hashes are unique but commit counts aren't: in the case of commit N being rolled back, the leader will immediately send out a new commit with the same commitCount N.  So there could conceivably be two different commits with the same commitCount N (but with different hashes), and one of them might be waiting to get the rollback message while the new commit gets sent out.

Reply:
It's like this because I initially started identifying commits by `commitCount`, but realized this wouldn't work when I added ROLLBACK later, so used hashes there. I never went back and cleaned up `_replicationHashes` to match, but you're right, they can both just be sets.